modules/caddyhttp/encode/encode.go

```diff
@@ -156,7 +156,7 @@ func (enc *Encode) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyh
 			if _, ok := enc.writerPools[encName]; !ok {
 				continue // encoding not offered
 			}
-			w = enc.openResponseWriter(encName, w)
+			w = enc.openResponseWriter(encName, w, r.Method == http.MethodConnect)
```

Comment:
We should try to unify the handling of h1 and h2 websockets. Maybe we can drop the workaround in `AcceptedEncodings` for h1 websockets and extend the pass-through detection here?

```go
r.Header.Get("Sec-WebSocket-Key") != "" || r.Method == http.MethodConnect
```

Reply:
`CONNECT` is not just for websockets, it's for creating a tunnel. It shouldn't be encoded if response is successful regardless of what's it's used.

For websocket, I don't know why this exists. Successful upgrade will bypass response writer anyway. I think there was once a protocol upgrade called derp used by tailscale that failed because `encode` saved the `101` status.

Reply:
> I think there was once a protocol upgrade called derp used by tailscale that failed because `encode` saved the `101` status.

Indeed: #5419